packages/common-grpc/src/service.js

```diff
@@ -537,27 +577,37 @@ GrpcService.encodeValue_ = function(value, options) {
       blobValue: value
     };
   } else if (is.object(value)) {
-    convertedValue = {
-      structValue: GrpcService.objToStruct_(value, options)
-    };
+    if (this.seenObjects.has(value)) {
+      // Circular reference.
+      if (!this.removeCircular) {
+        throw new Error(
+            'Object has a circular reference – cannot encode as a struct');
```

Comment:
Since I believe this error will be returned to the user, maybe:

"This object contains a circular reference. To automatically remove it, set `removeCircular` to true."